tests/v2/validation/provisioning/clustertemplates/provisioning_cluster_template_test.go

```diff
@@ -0,0 +1,335 @@
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	if c.provisioningConfig.RKE1KubernetesVersions == nil {
+		rke1Versions, err := kubernetesversions.ListRKE1AllVersions(c.client)
+		require.NoError(c.T(), err)
+
+		c.provisioningConfig.RKE1KubernetesVersions = []string{rke1Versions[len(rke1Versions)-1]}
+	}
+
+	if c.provisioningConfig.CNIs == nil {
+		c.provisioningConfig.CNIs = []string{cniCalico}
+	}
+
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) createRke1ClusterTemplateAndTemplateRevisions(clusterTemplateRevisionConfig clustertemplates.ClusterTemplateRevisionConfig, answers map[string]string, members []mgmt.Member) *clustertemplates.ClusterTemplateConfig {
+	clusterTemplate, err := rke1.CreateRkeTemplate(c.client, members)
+	require.NoError(c.T(), err)
+
+	clusterTemplateRevision, err := rke1.CreateRkeTemplateRevision(c.client, clusterTemplateRevisionConfig, clusterTemplate.ID)
+	require.NoError(c.T(), err)
+
+	clusterTemplateClusterConfig := new(clustertemplates.ClusterTemplateConfig)
+	clusterTemplateClusterConfig.RKE1TemplateID = clusterTemplate.ID
+	clusterTemplateClusterConfig.RKE1TemplateRevisionID = clusterTemplateRevision.ID
+
+	if answers != nil {
+		clusterTemplateClusterConfig.RKETemplateAnswers = &mgmt.Answer{
+			Values: answers,
+		}
+	}
+	return clusterTemplateClusterConfig
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) testCreateRke1ClusterWithTemplateAndVerify(client *rancher.Client, clusterTempClusterConfig *rke1.ClusterTemplateConfig) *mgmt.Cluster {
+
+	rke1Provider := provisioning.CreateRKE1Provider(c.provisioningConfig.NodeProviders[0])
+	nodeTemplate, err := rke1Provider.NodeTemplateFunc(client)
+	require.NoError(c.T(), err)
+
+	clusterConfig := clusters.ConvertConfigToClusterConfig(c.provisioningConfig)
+
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+	clusterConfig.NodePools = nodeAndRoles
+
+	clusterObject, err := provisioning.CreateProvisioningRKE1ClusterWithClusterTemplate(client, rke1Provider, clusterTempClusterConfig, clusterConfig, nodeTemplate)
+	require.NoError(c.T(), err)
+
+	clusterConfig.KubernetesVersion = c.provisioningConfig.RKE1KubernetesVersions[0]
+	provisioning.VerifyRKE1Cluster(c.T(), c.client, clusterConfig, clusterObject)
+
+	if clusterObject.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig != nil {
+		err := extensionsetcdsnapshot.CreateRKE1Snapshot(client, clusterObject.Name)
+		require.NoError(c.T(), err)
+
+		existingSnapshots, err := extensionsetcdsnapshot.GetRKE1Snapshots(client, clusterObject.ID)
+		etcdNodeCount, _ := etcdsnapshot.MatchNodeToAnyEtcdRole(client, clusterObject.ID)
+
+		_, err = provisioning.VerifySnapshots(client, clusterObject.Name, etcdNodeCount+len(existingSnapshots), true)
+		require.NoError(c.T(), err)
+	}
+
+	return clusterObject
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) TestProvisioningRKE1ClusterWithClusterTemplate() {
```

Comment:
`ProvisioningRKE1Cluster` is part of the suite name and can be removed from the test name. So is ClusterTemplate